### PR TITLE
⭐️ add auto discovery for all resolvers

### PR DESCRIPTION
**Pull request by imilchev**
Part of #171

### CODE REVIEW

File: motor/discovery/aws/resolver.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func (r *Resolver) Name() string {
 }
 
 func (r *Resolver) AvailableDiscoveryTargets() []string {
-	return []string{DiscoveryAll, DiscoveryInstances, DiscoverySSM}
+	return []string{DiscoveryAll, common.DiscoveryAuto, DiscoveryInstances, DiscoverySSM}
```
Comment (chris-rock) [inline motor/discovery/aws/resolver.go]: Just adding it to the list is not doing the trick, the code needs to be adapted to handle that value as well

Comment (imilchev) [reply motor/discovery/aws/resolver.go]: I am still confused about this one. What needs to happen? Right now if I look at the code (let's take AWS as an example) there are 2 conditionals for discovery flags. One for SSM and one for instances. If we want the `DiscoveryAuto` to keep the behaviour the same as what is now when we are not specifying any flags, then this change should be enough. If we want it to behave as `DiscoveryAll` then I need to modify the code.

File: apps/cnquery/cmd/run.go [hunk 2]
```diff
@@ -49,7 +50,7 @@ var execCmd = builder.NewProviderCommand(builder.CommandOpts{
 
 		cmd.Flags().String("path", "", "path to a local file or directory that the connection should use")
 		cmd.Flags().StringToString("option", nil, "addition connection options, multiple options can be passed in via --option key=value")
-		cmd.Flags().String("discover", "", "enables the discovery of nested assets. Supported are 'all|instances|host-instances|host-machines|container|container-images|pods|cronjobs|statefulsets|deployments|jobs|replicasets|daemonsets'")
+		cmd.Flags().String("discover", common.DiscoveryAuto, "enables the discovery of nested assets. Supported are 'all|instances|host-instances|host-machines|container|container-images|pods|cronjobs|statefulsets|deployments|jobs|replicasets|daemonsets'")
```
Comment (czunker) [inline apps/cnquery/cmd/run.go]: Although "auto" is the default, shouldn't it be part of the "Supported ..." list?

The same for `shell` and `explore` command.

File: motor/discovery/common/discovery.go [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package common
+
+const DiscoveryAuto = "auto"
```
Comment (czunker) [inline motor/discovery/common/discovery.go]: I think with this, we can remove/change L24 + L47 in `motor/discovery/k8s/resolver.go`

Comment (imilchev) [reply motor/discovery/common/discovery.go]: yup, that's right. I fixed that

File: apps/cnquery/cmd/run.go (hunk 2 above)
Comment (chris-rock) [reply apps/cnquery/cmd/run.go]: This needs to be applied to all commands that use the --discover flag

File: motor/discovery/aws/resolver.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ import (
 
 const (
 	DiscoveryAll       = "all"
+	DiscoveryAccounts  = "accounts"
 	DiscoveryInstances = "instances"
 	DiscoverySSM       = "ssm"
```
Comment (chris-rock) [inline motor/discovery/aws/resolver.go]: @vjeffrey should this not be better ssm-instances?

File: apps/cnquery/cmd/run.go (hunk 2 above)
Comment (imilchev) [reply apps/cnquery/cmd/run.go]: I noticed for `shell` the discovery list is much shorter than `run` and `exec`. Shouldn't the lists be identical?

**Pull request merged**